### PR TITLE
Fix crossgenning of framework libraries on Windows

### PR DESCRIPTION
**Pull request by trylek**
I tried to slightly improve the data flow by creating explicit
environment variables for the two arguments. There were a bunch
of errors regarding % vs !, duplicate double-quoting and the like.
It seems to be working now.

Thanks

Tomas

### CODE REVIEW

File: src/coreclr/build-test.cmd [hunk 1]
```diff
@@ -663,51 +663,65 @@ exit /b 0
...
+set AssemblyName=%2
+
 REM Skip mscorlib since it is already precompiled.
-if /I "%2" == "mscorlib.dll" exit /b 0
-if /I "%2" == "mscorlib.ni.dll" exit /b 0
+if /I "%AssemblyName%" == "mscorlib.dll" exit /b 0
+if /I "%AssemblyName%" == "mscorlib.ni.dll" exit /b 0
 REM don't precompile anything from CoreCLR
-if /I exist %CORE_ROOT_STAGE%\%2 exit /b 0
+if /I exist %CORE_ROOT_STAGE%\%AssemblyName% exit /b 0
```
Comment (janvorli) [inline src/coreclr/build-test.cmd]: Does this mean that we won't be crossgening System.Private.CoreLib.dll with crossgen2?

Comment (trylek) [reply src/coreclr/build-test.cmd]: It probably does and I concur it's something worth discussing moving forward just like the default compilation of S.P.C with legacy Crossgen within build.cmd/sh.

Comment (janvorli) [reply src/coreclr/build-test.cmd]: I think this is a bit different from using the new crossgen in build.cmd/sh. When we run the build-tests.cmd with the option asking for crossgen2 compilation, is it strange to get the whole framework crossgened with crossgen2 except for the S.P.C. I wonder if we could just add a check here that if we are using crossgen2, we would precompile the S.P.C. too.

Comment (trylek) [reply src/coreclr/build-test.cmd]: I agree it's weird. I basically left that in as a pre-existing conditional statement but thinking about this some more, what about throwing this away altogether? It doesn't make too much sense to me in the "new world".

Comment (janvorli) [reply src/coreclr/build-test.cmd]: That sounds reasonable.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

File: src/coreclr/build-test.cmd [hunk 2]
```diff
@@ -663,51 +663,59 @@ exit /b 0
...
+    set __CrossgenCmd=!__CrossgenExe! /Platform_Assemblies_Paths "!CORE_ROOT!" /in !AssemblyPath! /out !__CrossgenOutputFile! >nul 2>nul
 ) else (
-    "!CORE_ROOT_STAGE!\crossgen2\crossgen2 -r:"!CORE_ROOT!\*.dll" -O --inputbubble -out:"__CrossgenOutputFile" "%1" >nul 2>nul
+    set __CrossgenCmd=!__CrossgenExe! -r:"!CORE_ROOT!\*.dll" -O --inputbubble --out:!__CrossgenOutputFile! !AssemblyPath! >nul 2>nul
+    echo !__CrossgenCmd!
+    !__CrossgenCmd!
     set /a __exitCode = !errorlevel!
 )
 
+echo %__CrossgenCmd%
```
Comment (janvorli) [inline src/coreclr/build-test.cmd]: It seems this should be replaced  by the similar code from the `else` branch above and deleted from the `else` branch.

File: src/coreclr/build-test.cmd [hunk 3]
```diff
@@ -663,51 +663,59 @@ exit /b 0
...
+set __CrossgenOutputFile="%CORE_ROOT%\temp.ni._dll"
+set __CrossgenCmd=
 
 if defined __Crossgen (
-    "!__CrossgenExe!" /Platform_Assemblies_Paths "!CORE_ROOT!" /in "%1" /out "!__CrossgenOutputFile" >nul 2>nul
-    set /a __exitCode = !errorlevel!
+    set __CrossgenCmd=!__CrossgenExe! /Platform_Assemblies_Paths "!CORE_ROOT!" /in !AssemblyPath! /out !__CrossgenOutputFile! >nul 2>nul
 ) else (
-    "!CORE_ROOT_STAGE!\crossgen2\crossgen2 -r:"!CORE_ROOT!\*.dll" -O --inputbubble -out:"__CrossgenOutputFile" "%1" >nul 2>nul
+    set __CrossgenCmd=!__CrossgenExe! -r:"!CORE_ROOT!\*.dll" -O --inputbubble --out:!__CrossgenOutputFile! !AssemblyPath! >nul 2>nul
```
Comment (janvorli) [inline src/coreclr/build-test.cmd]: I think that to eliminate all the ugly warnings on bad image format of files that are .dll but not framework assemblies, we could replace `-r:"!CORE_ROOT!\*.dll` by `-r:"!CORE_ROOT!\System.*.dll -r:"!CORE_ROOT!\Windows.*.dll` I think this covers all the framework assemblies.

Comment (janvorli) [inline src/coreclr/build-test.cmd]: I suppose the `>nul 2>nul` was supposed to be part of the `__CrossgenCmd`, but it is applied to the `set` command instead. It should be placed on the `!__CrossgenCmd!` line if we want to suppress the crossgen output.

Comment (janvorli) [reply src/coreclr/build-test.cmd]: Oh, and `-r:!CORE_ROOT!\mscorlib.dll` too, some tests need that.

File: src/coreclr/build-test.cmd (hunk 2 above)
Comment (trylek) [reply src/coreclr/build-test.cmd]: Fixed, thanks for pointing that out!

File: src/coreclr/build-test.cmd (hunk 3 above)
Comment (trylek) [reply src/coreclr/build-test.cmd]: Good suggestion, thanks, fixed!

Comment (trylek) [reply src/coreclr/build-test.cmd]: Deleted per our offline chat, there's no sense in throwing away diagnostic information (and there's not a ton of it).

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**